src/test/TabooRoomTest.py

```diff
@@ -160,3 +164,160 @@ def testBasic(self):
                                         "score": [2]}],
                         {101, 102, 103}),
         ], anyOrder=True)
+
+class TabooWordSetTest(unittest.TestCase):
+    def testBasic(self):
+        wordset = SupportedWordSets["test"]
+
+        self.assertIsNone(wordset.nextWord({0, 1, 2}))
+        self.assertIsNone(wordset.nextWord({0, 1, 2, 3}))
+
+        def tryNextWord(usedWordIdxs, possibleValues):
+            found = wordset.nextWord(usedWordIdxs)
+            self.assertIn((found.word, found.disallowed), possibleValues)
+            self.assertEqual(len(found.usedWordIdxs - usedWordIdxs), 1)
+
+        tryNextWord({1, 2}, [("a", ["a1", "a2"])])
+        tryNextWord({0, 2}, [("b", ["b1", "b2"])])
+        tryNextWord({0, 1}, [("c", ["c1", "c2"])])
+
+        tryNextWord({0}, [("b", ["b1", "b2"]), ("c", ["c1", "c2"])])
+
+class TabooTurnManagerTest(unittest.TestCase, MsgTestLib):
+    def setUp(self):
+        random.seed(1)
+
+    def mockPlyrTeam(self, txq, teamId,
+                     connsByPlayerName,
+                     turnsPlayedByPlayerName=None):
+        turnsPlayedByPlayerName = turnsPlayedByPlayerName or {}
+
+        team = TabooTeam(txq, teamId)
+        for plyrName, conns in connsByPlayerName.items():
+            plyr = TabooPlayer(txq, name=plyrName, team=team)
+            for ws in conns:
+                plyr.playerConns.addConn(ws)
+                team.conns.addConn(ws)
```

Comment:
player.add|delConn(ws) should update team.conns as well. Then this line is not needed